### PR TITLE
fixlambda_runtime(sign): Update lambda runtimes to match current AWS guidelines

### PR DESCRIPTION
**Pull request by papaharry**
### Checklist
- [ ] Issue(s) linked

Haven't created an issue yet, but happy to if it needs it for the check

### CODE REVIEW

File: conformance_pack/lambda.sp [hunk 1]
```diff
@@ -365,12 +365,12 @@ query "lambda_function_use_latest_runtime" {
       arn as resource,
       case
         when package_type <> 'Zip' then 'skip'
-        when runtime in ('nodejs16.x', 'nodejs14.x', 'nodejs12.x', 'nodejs10.x', 'python3.9', 'python3.8', 'python3.7', 'python3.6', 'ruby2.5', 'ruby2.7', 'java11', 'java8', 'java8.al2', 'go1.x', 'dotnetcore2.1', 'dotnetcore3.1', 'dotnet6') then 'ok'
+        when runtime in (''nodejs18.x', 'nodejs16.x', 'nodejs14.x', 'python3.10', 'python3.9', 'python3.8', 'python3.7', 'ruby3.2', 'ruby2.7', 'java17', 'java11', 'java8', 'go1.x', 'dotnet7', 'dotnet6') then 'ok'
```
Comment (rajlearner17) [inline conformance_pack/lambda.sp:368]:
```suggestion
        when runtime in ('nodejs18.x', 'nodejs16.x', 'nodejs14.x', 'python3.10', 'python3.9', 'python3.8', 'python3.7', 'ruby3.2', 'ruby2.7', 'java17', 'java11', 'java8', 'go1.x', 'dotnet7', 'dotnet6') then 'ok'
```

Comment (papaharry) [reply conformance_pack/lambda.sp]: Oops. Accepted the suggestion. Thank you!

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Comment by khushboo9024** [comment]
Hi [papaharry](https://github.com/papaharry) thanks for raising the PR. 👍 

We noticed that the list of runtimes in the query does not match the runtimes specified in the FSBP [document](https://docs.aws.amazon.com/securityhub/latest/userguide/lambda-controls.html#lambda-2), which we strictly follow for all FSBP controls.

Could you kindly update the query to align with the specifications mentioned in the FSBP document? Alternatively, please provide the source document you referenced for this information.

image

Thanks.

**Comment by papaharry** [comment]
Hello, I have added the missing runtime. I took the runtimes from the [here](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) but I missed that one.

**Review by misraved — APPROVED** [review]

**Pull request merged**